examples/hosts/app-integration/external-data/src/model/taskList.ts

```diff
@@ -0,0 +1,238 @@
+import { TypedEventEmitter } from "@fluidframework/common-utils";
+import { SharedString } from "@fluidframework/sequence";
+
+import { externalDataSource, parseStringData } from "../externalData";
+import type { ITask, ITaskEvents, ITaskList } from "../modelInterfaces";
+
+class Task extends TypedEventEmitter<ITaskEvents> implements ITask {
+    public get id() {
+        return this._id;
+    }
+    // Probably would be nice to not hand out the SharedString, but the CollaborativeInput expects it.
+    public get name() {
+        return this._name;
+    }
+    public get priority() {
+        const cellValue = this._priority.get();
+        if (cellValue === undefined) {
+            throw new Error("Expected a valid priority");
+        }
+        return cellValue;
+    }
+    public set priority(newValue: number) {
+        this._priority.set(newValue);
+    }
+    public constructor(
+        private readonly _id: string,
+        private readonly _name: SharedString,
+        private readonly _priority: SharedCell<number>,
+    ) {
+        super();
+        this._name.on("sequenceDelta", () => {
+            this.emit("nameChanged");
+        });
+        this._priority.on("valueChanged", () => {
+            this.emit("priorityChanged");
+        });
+    }
+}
+
+/**
+ * The TaskList is our data object that implements the ITaskList interface.
+ */
+export class TaskList extends DataObject implements ITaskList {
+    /**
+     * The tasks collection holds local facades on the data.  These facades encapsulate the data for a single task
+     * so we don't have to hand out references to the whole SharedDirectory.  Additionally, we only create them
+     * after performing the async operations to retrieve the constituent data (.get() the handles) which allows
+     * the ITask interface to be synchronous, and therefore easier to use in a view.  This is an example of the
+     * collection pattern -- see the contact-collection example for more details on this pattern.
+     */
+    private readonly tasks = new Map<string, Task>();
+
+    public readonly addTask = (id: string, name: string, priority: number) => {
+        if (this.tasks.get(id) !== undefined) {
+            throw new Error("Task already exists");
+        }
+        const nameString = SharedString.create(this.runtime);
+        nameString.insertText(0, name);
+        const priorityCell: SharedCell<number> = SharedCell.create(this.runtime);
+        priorityCell.set(priority);
+        // To add a task, we update the root SharedDirectory.  This way the change is propagated to all collaborators
+        // and persisted.  In turn, this will trigger the "valueChanged" event and handleTaskAdded which will update
+        // the this.tasks collection.
+        this.root.set(id, { id, name: nameString.handle, priority: priorityCell.handle });
+        // TODO: Ultimately we want to retain the data we retrieved from the external source separate from the draft
```

Comment:
@WayneFerrao I think this will be the best TODO to start on -- maybe we can chat about this a bit tomorrow too?